andela/ah-infinity-stones
# #161966910 Validate User Registration

#### What does this PR do?
- Implement user registration validation
#### Description of Task to be completed?
- Use regex to ensure data passed by the user meets the criteria set.
#### How should this be manually tested?
1. `git clone -b ft-user-signup-validation-161966910 https://github.com/andela/ah-infinity-stones` and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_NAME='ah_dev'
export DATABASE_USER='user'
export DATABASE_PASSWORD='password'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run tests - `$ python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/users/` provide inputs in the following format and submit.
```
{ "user": {
"username": "username44",
"email": "",
"password": ""	
}
}
```
*Note* : Provide username that's less than 4 characters or an invalid email or a password that doesn't have a special character, just to see the validations in action.
#### What are the relevant pivotal tracker stories?
[#161966910](https://www.pivotaltracker.com/story/show/161966910)
#### Screenshots
<img width="965" alt="screenshot 2018-12-11 at 17 04 03" src="https://user-images.githubusercontent.com/12385482/49805718-e2e5c280-fd66-11e8-9a75-93341efdf8af.png">
